emcc.py

```diff
@@ -1896,8 +1896,10 @@ def phase_linker_setup(options, state, newargs):
   if settings.LZ4:
     settings.EXPORTED_RUNTIME_METHODS += ['LZ4']
 
+  # WASM_BIGINT defaults to 1 when VMs support it, and when not using wasm2js
   default_setting('WASM_BIGINT',
-                  feature_matrix.caniuse(feature_matrix.Feature.JS_BIGINT_INTEGRATION))
+                  feature_matrix.caniuse(feature_matrix.Feature.JS_BIGINT_INTEGRATION) and \
```

Comment:
Can we have caniuse take into account settings.WASM2JS perhaps?

Reply:
How were you thinking to use it? That when wasm2js, we don't need newer browsers? That might work as long as wasm2js doesn't use newer features, but I'm not sure that will stay that way...

Reply:
I mean why doesn't `feature_matrix.caniuse(feature_matrix.Feature.JS_BIGINT_INTEGRATION)`  just return false if `settings.WASM2JS`?  (making this condition here simpler).

Reply:
I thought that function is focused just on the feature matrix itself, that is, comparing browser versions? Maybe it's the name that's confusing me. We could rename it to `caniuse.caniuse` to make it clear it's making the entire decision?

Reply:
Sure I guess keeping `feature_matrix.caniuse` pure makes sense.